flink-core/src/main/java/org/apache/flink/core/io/VersionedIOReadableWritable.java

```diff
@@ -77,6 +77,11 @@ private void resolveVersionRead(int readVersion) throws VersionMismatchException
 			}
 		}
 
+		// TODO this is a temporary workaround for FLINK-9377 that should be removed
+		if (readVersion == getVersion() + 1) {
```

Comment:
This seems fragile here, because it is in a common base class outside the serializer utils.
Can this be solved by overriding "getCompatibleVersions()" instead?